src/plugins/ui_actions/public/triggers/trigger_internal.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Trigger, TriggerContext, TriggerContextParams } from './trigger';
+import { TriggerContract } from './trigger_contract';
+import { UiActionsService } from '../service';
+import { Action } from '../actions';
+import { buildContextMenuForActions, openContextMenu } from '../context_menu';
+
+/**
+ * Internal representation of a trigger kept for consumption only internally
+ * within `ui_actions` plugin.
+ */
+export class TriggerInternal<T extends Trigger<any, any>> {
+  public readonly contract = new TriggerContract<T>(this);
+
+  public get id() {
+    return this.trigger.id;
```

Comment:
Is this providing any functionality?

Reply:
I've removed this getter, using the `trigger.id` instead.